packages/module/src/Hooks/pagination.ts

```diff
@@ -1,31 +1,84 @@
-import { useState } from "react";
+import { useState, useEffect } from "react";
 
 export interface UseDataViewPaginationProps {
   /** Initial page */
   page?: number;
   /** Items per page */
   perPage: number;
+  /** Current search parameters as a string */
+  searchParams?: URLSearchParams;
+  /** Function to set search parameters */
+  setSearchParams?: (params: URLSearchParams) => void;
 }
 
 export interface DataViewPaginationProps extends UseDataViewPaginationProps {
   /** Current page number */
-    page: number;
+  page: number;
 }
 
-export const useDataViewPagination = ({ page = 1, perPage }: UseDataViewPaginationProps) => { 
-  const [ state, setState ] = useState({ page, perPage });
-  
-  const onPerPageSelect = (_event: React.MouseEvent | React.KeyboardEvent | MouseEvent | undefined, newPerPage: number) => {
-    setState(prev => ({ ...prev, perPage: newPerPage }));
-  }
-  
-  const onSetPage = (_event: React.MouseEvent | React.KeyboardEvent | MouseEvent | undefined, newPage: number) => {
+export const useDataViewPagination = ({
+  page = 1,
+  perPage,
+  searchParams,
+  setSearchParams,
+}: UseDataViewPaginationProps) => {
+  const [ state, setState ] = useState({
+    page: searchParams?.get('page') !== null
```

Comment:
I think you want some enum or constants for the query param names. Just so you don't have string around the code which might get tricky to update at some point.